### PR TITLE
JUnit 5 - Parameterize tests using annotation @ParameterizedTest

### PR DESCRIPTION
**Pull request by adavid9**
Partial fix for #929
This is a few tests that have been parameterized.

### CODE REVIEW

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 1]
```diff
@@ -33,80 +35,57 @@ public void init()
...
-               .when()
-               .get(getContextBaseUrl())
-               .peek()
-               .then()
-               .statusCode(HttpStatus.OK.value())
-               .body(equalTo("pong"));
-    }
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    void shouldReturnPong(String acceptHeader) {
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Cosmetic: Could you please, follow the project's coding convention?

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 2]
```diff
@@ -33,80 +35,57 @@ public void init()
...
-               .when()
-               .get(getContextBaseUrl() + "/token")
-               .peek()
-               .then()
-               .statusCode(HttpStatus.OK.value())
-               .body("message", equalTo("pong"));
-    }
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    void shouldReturnPongForAuthenticatedUsers(String acceptHeader) {
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 3]
```diff
@@ -33,80 +35,57 @@ public void init()
...
-    {
-        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
-               .when()
-               .get(getContextBaseUrl() + "/token")
-               .peek()
-               .then()
-               .log().all()
-               .statusCode(HttpStatus.UNAUTHORIZED.value())
-               .body(notNullValue());
+    void anonymousUsersShouldNotBeAbleToAccess(String acceptHeader) {
```
Comment (carlspring) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Same as above.

**Comment by carlspring** [comment]
Could you please follow the project's [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html)? For more details, please check [here](https://strongbox.github.io/contributing.html) as well.

**Comment by adavid9** [comment]
In the next `PR` I will add commit with `fixed typo`

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 4]
```diff
@@ -79,12 +82,20 @@ void anonymousUsersShouldNotBeAbleToAccess(String acceptHeader) {
                 .body(notNullValue());
     }
 
-    private void getResponseBody(ValidatableMockMvcResponse response, String acceptHeader, String message) {
-        if (acceptHeader.equals(MediaType.APPLICATION_JSON_VALUE)) {
+    private void getResponseBody(ValidatableMockMvcResponse response,
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Please, change this method name to `validateResponseBody`.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerTest.java [hunk 5]
```diff
@@ -204,12 +204,20 @@ public void testDeleteArtifactAndEmptyTrashForAllRepositories(String acceptHeade
                     "Failed to empty trash for repository '" + REPOSITORY_WITH_TRASH + "'!");
     }
 
-    private void getResponseBody(ValidatableMockMvcResponse response, String acceptHeader, String message) {
-        if (acceptHeader.equals(MediaType.APPLICATION_JSON_VALUE)) {
+    private void getResponseBody(ValidatableMockMvcResponse response,
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerTest.java]: Same as previous comment.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerUndeleteTest.java [hunk 6]
```diff
@@ -241,12 +243,20 @@ void testUndeleteArtifactsForAllRepositories(String acceptHeader) throws Excepti
                                     "+p:jar");
     }
 
-    private void getResponseBody(ValidatableMockMvcResponse response, String acceptHeader, String message) {
-        if (acceptHeader.equals(MediaType.APPLICATION_JSON_VALUE)) {
+    private void getResponseBody(ValidatableMockMvcResponse response,
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerUndeleteTest.java]: Same as previous comment.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 7]
```diff
@@ -154,32 +155,56 @@ void testSetAndGetNumberOfConnectionsForProxyRepositoryWithTextAcceptHeader(Stri
         getResponseBodyPoolStats(acceptHeader, response, expectedPoolStats);
     }
 
-    private void getResponseBodyConnections(ValidatableMockMvcResponse response, String acceptHeader, int newMaxNumberOfConnections) {
-        if (acceptHeader.equals(MediaType.APPLICATION_JSON_VALUE)) {
+    private void getResponseBodyConnections(ValidatableMockMvcResponse response,
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Please, change this method name to `validateResponseBodyConnections`.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 8]
```diff
@@ -154,32 +155,56 @@ void testSetAndGetNumberOfConnectionsForProxyRepositoryWithTextAcceptHeader(Stri
...
+        }
+        else
+        {
             throw new IllegalArgumentException("Unsupported content type: " + acceptHeader);
         }
     }
 
-    private void getResponseBody(ValidatableMockMvcResponse response, String acceptHeader, String message) {
-        if (acceptHeader.equals(MediaType.APPLICATION_JSON_VALUE)) {
+    private void getResponseBody(ValidatableMockMvcResponse response,
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as previous comment.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 9]
```diff
@@ -154,32 +155,56 @@ void testSetAndGetNumberOfConnectionsForProxyRepositoryWithTextAcceptHeader(Stri
...
+        }
+        else
+        {
             throw new IllegalArgumentException("Unsupported content type: " + acceptHeader);
         }
     }
 
-    private void getResponseBodyPoolStats(String acceptHeader, ValidatableMockMvcResponse response, PoolStats expectedPoolStats) {
-        if (acceptHeader.equals(MediaType.APPLICATION_JSON_VALUE)) {
+    private void getResponseBodyPoolStats(String acceptHeader,
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Please, change this method name to `validateResponseBodyPoolStats`.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java [hunk 10]
```diff
@@ -170,7 +174,8 @@ void roleShouldNotBeDeleted(String acceptHeader)
 
     @ParameterizedTest
     @MethodSource("privilegesProvider")
-    void privilegesToAnonymousShouldBeAdded(String privilegeName, String acceptHeader) {
+    void privilegesToAnonymousShouldBeAdded(String privilegeName, String acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java]: Cosmetic: Could you please, follow the project's coding convention for the method parameters?

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 11]
```diff
@@ -33,80 +35,68 @@ public void init()
...
     }
 
-    @Test
-    public void testShouldReturnPongText()
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    void shouldReturnPong(String acceptHeader)
     {
+        ValidatableMockMvcResponse response = given()
+                .header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Please, use `.accept(acceptHeader)` instead.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 12]
```diff
@@ -33,80 +35,68 @@ public void init()
...
-        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
-               .when()
-               .get(getContextBaseUrl() + "/token")
-               .peek()
-               .then()
-               .statusCode(HttpStatus.OK.value())
-               .body("message", equalTo("pong"));
-    }
+        ValidatableMockMvcResponse response = given()
+                .header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java [hunk 13]
```diff
@@ -33,80 +35,68 @@ public void init()
...
     {
-        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
-               .when()
-               .get(getContextBaseUrl() + "/token")
-               .peek()
-               .then()
-               .log().all()
-               .statusCode(HttpStatus.UNAUTHORIZED.value())
-               .body(notNullValue());
+        given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/PingControllerTest.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerTest.java [hunk 14]
```diff
@@ -154,18 +157,21 @@ public void testForceDeleteArtifactAllowed()
...
-    @Test
-    public void testDeleteArtifactAndEmptyTrashForRepositoryWithTextAcceptHeader()
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    void testDeleteArtifactAndEmptyTrashForRepository(String acceptHeader)
     {
         String url = getContextBaseUrl() + "/api/trash/" + STORAGE0 + "/" + REPOSITORY_WITH_TRASH;
 
-        given().header(HttpHeaders.ACCEPT, MediaType.TEXT_PLAIN_VALUE)
+        ValidatableMockMvcResponse response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerTest.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerTest.java [hunk 15]
```diff
@@ -177,55 +183,43 @@ private Path getPathToArtifactInTrash()
...
-    @Test
-    public void testDeleteArtifactAndEmptyTrashForAllRepositoriesWithTextAcceptHeader()
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    void testDeleteArtifactAndEmptyTrashForAllRepositories(String acceptHeader)
     {
         String url = getContextBaseUrl() + "/api/trash";
 
-        given().header(HttpHeaders.ACCEPT, MediaType.TEXT_PLAIN_VALUE)
+        ValidatableMockMvcResponse response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerTest.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerUndeleteTest.java [hunk 16]
```diff
@@ -211,13 +215,15 @@ public void testUndeleteArtifactsForAllRepositoriesWithTextAcceptHeader()
 
         String url = getContextBaseUrl() + "/api/trash";
 
-        given().header(HttpHeaders.ACCEPT, MediaType.TEXT_PLAIN_VALUE)
+        ValidatableMockMvcResponse response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/TrashControllerUndeleteTest.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 17]
```diff
@@ -43,115 +45,70 @@ public void init()
...
         String url = getContextBaseUrl() + "/api/configuration/proxy/connection-pool/max/" + newMaxNumberOfConnections;
 
-        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
-               .when()
-               .put(url)
-               .peek()
-               .then()
-               .statusCode(HttpStatus.OK.value())
-               .body("message", equalTo("Max number of connections for proxy repository was updated successfully."));
+        ValidatableMockMvcResponse response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 18]
```diff
@@ -43,115 +45,70 @@ public void init()
...
 
-        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
-               .when()
-               .get(url)
-               .peek()
-               .then()
-               .statusCode(HttpStatus.OK.value())
-               .body("numberOfConnections", equalTo(newMaxNumberOfConnections));
+        url = getContextBaseUrl() + "/api/configuration/proxy/connection-pool";
+        response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 19]
```diff
@@ -43,115 +45,70 @@ public void init()
...
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    public void testSetAndGetDefaultNumberOfConnectionsForProxyRepository(String acceptHeader)
     {
         int newDefaultNumberOfConnections = 5;
 
         String url =
                 getContextBaseUrl() + "/api/configuration/proxy/connection-pool/default/" + newDefaultNumberOfConnections;
 
-        given().header(HttpHeaders.ACCEPT, MediaType.TEXT_PLAIN_VALUE)
+        ValidatableMockMvcResponse response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 20]
```diff
@@ -43,115 +45,70 @@ public void init()
...
-               .statusCode(HttpStatus.OK.value())
-               .body("message",
-                     equalTo("Default number of connections for proxy repository was updated successfully."));
+        String message = "Default number of connections for proxy repository was updated successfully.";
+        getResponseBody(response, acceptHeader, message);
 
         url = getContextBaseUrl() + "/api/configuration/proxy/connection-pool/default-number";
 
-        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+        response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 21]
```diff
@@ -173,70 +130,82 @@ public void testSetAndGetNumberOfConnectionsForProxyRepositoryWithTextAcceptHead
                      repository.getId() + "/" +
                      numberOfConnections;
 
-        given().header(HttpHeaders.ACCEPT, MediaType.TEXT_PLAIN_VALUE)
+        ValidatableMockMvcResponse response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java [hunk 22]
```diff
@@ -173,70 +130,82 @@ public void testSetAndGetNumberOfConnectionsForProxyRepositoryWithTextAcceptHead
...
+        String message = "Number of pool connections for repository was updated successfully.";
+        getResponseBody(response, acceptHeader, message);
 
         url = getContextBaseUrl() + "/api/configuration/proxy/connection-pool/" +
               repository.getStorage().getId() + "/" +
               repository.getId();
 
         PoolStats expectedPoolStats = new PoolStats(0, 0, 0, numberOfConnections);
-        given().header(HttpHeaders.ACCEPT, MediaType.TEXT_PLAIN_VALUE)
+        response = given().header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/HttpConnectionPoolConfigurationManagementControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java [hunk 23]
```diff
@@ -110,39 +101,33 @@ private void roleShouldNotBeAdded(String acceptHeader,
...
+        customRole.setName(roleName);
 
-    @Test
-    public void testEmptyRoleNameShouldNotBeAddedWithResponseInText()
-    {
-        String roleName = "";
-        roleShouldNotBeAdded(MediaType.TEXT_PLAIN_VALUE, roleName);
+        given()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java [hunk 24]
```diff
@@ -110,39 +101,33 @@ private void roleShouldNotBeAdded(String acceptHeader,
...
+                .body(containsString(FAILED_ADD_ROLE));
     }
 
-    private void configXMLCouldBeDownloaded(String acceptHeader)
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
+    void configXMLCouldBeDownloaded(String acceptHeader)
     {
         given().contentType(MediaType.APPLICATION_JSON_VALUE)
                .header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java]: Same as above.

File: strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java [hunk 25]
```diff
@@ -153,21 +138,12 @@ private void configXMLCouldBeDownloaded(String acceptHeader)
...
-                                     String roleName)
+    @ParameterizedTest
+    @ValueSource(strings = { MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE })
+    void roleShouldBeDeleted(String acceptHeader)
     {
+        // get role name
+        String roleName = config.getRoles().iterator().next().getName();
         // delete role
         given().contentType(MediaType.APPLICATION_JSON_VALUE)
                .header(HttpHeaders.ACCEPT, acceptHeader)
```
Comment (ptirador) [inline strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/configuration/security/authorization/AuthorizationConfigControllerTestIT.java]: Same as above.

**Comment by carlspring** [comment]
@ptirador ,

Thanks for reviewing!

@adavid9 ,

Congratulations on your first pull request! :)

**Pull request merged**